### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by ashleybarrett**
Implemented Jailbroken method and tests

### CODE REVIEW

File: Source/Extensions/NSError+Judo.h [hunk 1]
```diff
@@ -72,6 +73,7 @@ typedef NS_ENUM(NSUInteger, JudoError) {
     JudoErrorParameterError,
     JudoErrorResponseParseError,
     JudoErrorInvalidCardNumberError,
+    JudoJailbrokenDeviceDisallowed,
```
Comment (Ryce) [inline Source/Extensions/NSError+Judo.h]: should be JudoErrorJailbrokenDeviceDisallowed - consistency

File: Source/Extensions/NSError+Judo.m [hunk 2]
```diff
@@ -40,13 +40,18 @@
 NSString * const ErrorUserDidCancel = @"Received when user cancels the payment journey";
 NSString * const ErrorParameterError = @"A parameter entered into the dictionary (request body to Judo API) is not set";
 NSString * const ErrorFailed3DSRequest = @"After receiving the 3DS payload, when the payload has faulty data, the WebView fails to load the 3DS Page or the resolution page";
+NSString * const JailbrokenDeviceDisallowed = @"The device the code is currently running is jailbroken. Jailbroken devices are not allowed when instantiating a new Judo session";
```
Comment (Ryce) [inline Source/Extensions/NSError+Judo.m]: need to confirm with Niklas, can not just add this

File: Source/JudoKit.m [hunk 3]
```diff
@@ -62,13 +62,29 @@ @interface JudoKit ()
...
 }
 
 - (instancetype)initWithToken:(NSString *)token secret:(NSString *)secret allowJailbrokenDevices:(BOOL)jailbrokenDevicesAllowed {
     self = [super init];
     if (self) {
+        
+        // Check if device is jailbroken and SDK was set to restrict access.
+        //Self is returned here without setting the token and secret. This will fail later.
+        if (!jailbrokenDevicesAllowed && [self isCurrentDeviceJailbroken]) {
+            return self;
```
Comment (Ryce) [inline Source/JudoKit.m]: add a comment here which describes what returning self without setting the headers will result in.

File: Tests/APITests/ErrorMessageContentsTests.swift [hunk 4]
```diff
@@ -19,6 +19,7 @@ let ErrorResponseParseError = "An error with a response from the backend API"
 let ErrorUserDidCancel = "Received when user cancels the payment journey"
 let ErrorParameterError = "A parameter entered into the dictionary (request body to Judo API) is not set"
 let ErrorFailed3DSRequest = "After receiving the 3DS payload, when the payload has faulty data, the WebView fails to load the 3DS Page or the resolution page"
+let JailbrokenDeviceDisallowed = "The device the code is currently running is jailbroken. Jailbroken devices are not allowed when instantiating a new Judo session";
```
Comment (Ryce) [inline Tests/APITests/ErrorMessageContentsTests.swift]: adjust with the above comment

File: Source/Extensions/NSError+Judo.h (hunk 1 above)
Comment (ashleybarrett) [inline Source/Extensions/NSError+Judo.h]: changed.

File: Source/Extensions/NSError+Judo.m (hunk 2 above)
Comment (ashleybarrett) [inline Source/Extensions/NSError+Judo.m]: This was copied from JudoKit.

File: Source/JudoKit.m (hunk 3 above)
Comment (ashleybarrett) [inline Source/JudoKit.m]: changed.

File: Tests/APITests/ErrorMessageContentsTests.swift (hunk 4 above)
Comment (ashleybarrett) [inline Tests/APITests/ErrorMessageContentsTests.swift]: Copied from JudoKit.

File: Source/Extensions/NSError+Judo.m (hunk 2 above)
Comment (Ryce) [inline Source/Extensions/NSError+Judo.m]: and change the JailbrokenDeviceDisallowed string to `ErrorJailbrokenDeviceDisallowed` for consistency, or change all others

File: Tests/APITests/ErrorMessageContentsTests.swift (hunk 4 above)
Comment (Ryce) [inline Tests/APITests/ErrorMessageContentsTests.swift]: same here as above

File: Source/Extensions/NSError+Judo.m (hunk 2 above)
Comment (ashleybarrett) [inline Source/Extensions/NSError+Judo.m]: changed,

File: Tests/APITests/ErrorMessageContentsTests.swift (hunk 4 above)
Comment (ashleybarrett) [inline Tests/APITests/ErrorMessageContentsTests.swift]: done.

**Comment by Ryce** [comment]
👍

**Pull request merged**